tracks/python/exercises/say/mentoring.md

```diff
@@ -0,0 +1,127 @@
+
+The main challenges are ensuring that the 'hundred' denominator is conditionally included and that the tens and ones columns are joined with a hyphen if they are both present.
+Simple `if` checks for each of these conditions are the most straightforward approach.
+
+Another valid approach is to only handle values below 100, and instead handle the hundreds alongside the other denominators.
+This approach typically requires the use of recursion. 
+
+### Larger numbers
+Encourage the student to use the first function they wrote for each set of three digits rather than duplicating code.
+Each set of three can be checked individually and conditionally added along with the appropriate denominator, or alternatively the denominators can be stored in a list and iterated over.
+
+### Zero or invalid input
+Finally, the student must add a guard that returns a `ValueError` if the input is out of range. 
+A guard is also the easiest way to handle the zero case, but this can also be handled through short circuit evaluation.
+
+### Parsing the number
+The number needs to be broken up into groups of three. 
+This can be handled either with numeric operators or by casting to a string.
+Generally a student will find it easier to work with the number as a string for this problem, but both approaches are valid.
+
+## Talking points
+The student should be familiar with comprehensions - if they aren't, introduce the concept as they are very useful here.
+
+There are a wide variety of valid approaches to this problem, which gives students a lot of opportunity to experiment and encounter new ways of manipulating data.
+Some useful techniques that students might not have encountered are listed below.
+Students will not need or want to use all of these, but mentors should look for opportunities in the student's code to introduce one or more of them.
+* Casting between numeric types and strings
+* String indexing, iteration, formatting, or concatenation
+* String methods such as `zfill`, `isnumeric`, `split`, or `join`
+* List concatenation using the `+` operator
+* List methods such as `append` or `extend`
+* Iterator functions such as `zip`, `enumerate`, or `filter`
+* Assignment operators
+* Unpacking operator
+* Short circuit evaluation and the use of `or`/`and` for control flow
+* Ternery if-else
+* Integer division and modulo operators, and the divmod function
+* Recusion
+
+Students who wish to extend the exercise can rewrite the function to include 'and' as a seperator as well as handle negative or larger numbers.
+This will require rewriting the tests, which might also serve as a learning opportunity for the student.
+
+This problem is a good opportunity to practice writing concise code and balancing readability. 
+It is also a good chance to code verbosely while the problem is solved, and then iterate on the solution to reduce length and complexity, so don't require the student to write concise code on every iteration. 
```

Comment:
```suggestion
It is also a good chance to code verbosely while the problem is solved, and then iterate on the solution to reduce length and complexity.
```

Avoid telling the mentor to do things or not do things. Provide discussion points and concepts which are useful for solving the exercise.